docs/plugins/requirements/image-processing.md

```diff
@@ -93,6 +93,189 @@ The following environments come with a preinstalled version of [Cairo Graphics]:
   [Docker image]: https://hub.docker.com/r/squidfunk/mkdocs-material/
   [GitHub Actions]: ../../publishing-your-site.md#with-github-actions
 
+#### Troubleshooting { id="cairosvg-troubleshooting" }
+
+After following the installation guide above it may happen that you still
+get the following error:
+
+```bash
+no library called "cairo-2" was found
+no library called "cairo" was found
+no library called "libcairo-2" was found
+cannot load library 'libcairo.so.2': error 0x7e.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libcairo.so.2'
+cannot load library 'libcairo.2.dylib': error 0x7e.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libcairo.2.dylib'
+cannot load library 'libcairo-2.dll': error 0x7e.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libcairo-2.dll'
+```
+
+This means that the [`cairosvg`][PyPi CairoSVG] package was installed,
+but the underlying [`cairocffi`][PyPi CairoCFFI] dependency couldn't
+[find][cffi-dopen] the installed library. Depending on the operating system
+the library lookup process is different:
+
+!!! tip
+    Before proceeding remember to fully restart any open Terminal windows, and
+    their parent hosts like IDEs to reload any environmental variables, which
+    were altered during the installation process. This might be the quick fix.
+
+=== ":material-apple: macOS"
+
+    On macOS the library lookup checks inside paths defined in [dyld].
+    Additionally each library `name` is [checked][find-library-macOS] in three
+    variants with the `libname.dylib`, `name.dylib` and `name.framework/name`
+    format.
+
+    [Homebrew] should set every needed variable to point at the installed
+    library, but if that didn't happen, you can use the debug script below
+    to maybe find the issue.
+
+    A [known workaround][cffi-issue] is to add the homebrew lib path
+    directly before running MkDocs:
+
+    ```bash
+    export DYLD_FALLBACK_LIBRARY_PATH=/opt/homebrew/lib
+    ```
+
+    ??? tip "Python Debug macOS Script"
```

Comment:
Just an idea: we _could_ create a `etc/debug` folder in our repository and move the script there. That way we can version it + we could, if necessary, use Snippets to integrate it. In the future, we might also somehow move this logic into the social plugin (or better: into our own utilities folder, so we can use it in other plugins as well) and/or bundle the script the theme (not now, let's keep it simple for now).

What do you think? I'm not sure if this makes sense, just wanted to bounce ideas.

Reply:
There is already the https://github.com/squidfunk/mkdocs-material/tree/master/includes directory. We could move the scripts there into the `etc/debug` structure, and later include them with Snippets, as you said. 

I'm still not sure about the Linux scripts, because GitHub Actions finds the library with https://www.unix.com/man-page/Linux/1/ld/ but I'm not sure if that will always be the case with the cairo library or if it depends on how someone installs it 😑 

> social plugin (or better: into our own utilities folder, so we can use it in other plugins as well) and/or bundle the script the theme (not now, let's keep it simple for now).

The scripts I provided are supposed to display the paths that are checked or the way they're checked for the user that wants to troubleshoot them. They are somewhat generalized, since they run with the `ctypes.util` built-in and not cairoffi directly, but I'm doubtful they would be useful as utils in the plugins. 

Another idea I had, was to ask you to create a mkdocs-material/troubleshooting repository and link to it from that troubleshooting guide, and there would be CI similar to my own repository, which people could copy to execute the actions and see the scripts in action, but this idea is a bit "out there" doesn't really focus on the matter at hand of getting it to work on the users PC, and requires "GitHub Actions proficiency" of the user.

Reply:
Yes, let's move it into `include/debug` and use Snippets for now.

> Another idea I had, was to ask you to create a mkdocs-material/troubleshooting repository and link to it from that troubleshooting guide, and there would be CI similar to my own repository, which people could copy to execute the actions and see the scripts in action, but this idea is a bit "out there" doesn't really focus on the matter at hand of getting it to work on the users PC, and requires "GitHub Actions proficiency" of the user.

Interesting idea, I'll discuss it with the core team next time we speak ☺️